src/cli.js

```diff
@@ -235,41 +317,58 @@ function cli(inputArgs) {
         }
 
-        cordova.raw[cmd].call(null, opts.platforms)
-        .then(function (platformChecks) {
+        var result = cordova.raw[cmd].call(null, opts.platforms)
+            .then(function(platformChecks) {
 
-            var someChecksFailed = Object.keys(platformChecks).map(function (platformName) {
-                events.emit('log', '\nRequirements check results for ' + platformName + ':');
-                var platformCheck = platformChecks[platformName];
-                if (platformCheck instanceof CordovaError) {
-                    events.emit('warn', 'Check failed for ' + platformName + ' due to ' + platformCheck);
-                    return true;
-                }
-
-                var someChecksFailed = false;
-                platformCheck.forEach(function (checkItem) {
-                    var checkSummary = checkItem.name + ': ' +
-                                    (checkItem.installed ? 'installed ' : 'not installed ') +
-                                    (checkItem.metadata.version || '');
-                    events.emit('log', checkSummary);
-                    if (!checkItem.installed) {
-                        someChecksFailed = true;
-                        events.emit('warn', checkItem.metadata.reason);
+                var someChecksFailed = Object.keys(platformChecks).map(function(platformName) {
+                    events.emit('log', '\nRequirements check results for ' + platformName + ':');
+                    var platformCheck = platformChecks[platformName];
+                    if (platformCheck instanceof CordovaError) {
+                        events.emit('warn', 'Check failed for ' + platformName + ' due to ' + platformCheck);
+                        return true;
                     }
+
+                    var someChecksFailed = false;
+                    platformCheck.forEach(function(checkItem) {
+                        var checkSummary = checkItem.name + ': ' +
+                            (checkItem.installed ? 'installed ' : 'not installed ') +
+                            (checkItem.metadata.version || '');
+                        events.emit('log', checkSummary);
+                        if (!checkItem.installed) {
+                            someChecksFailed = true;
+                            events.emit('warn', checkItem.metadata.reason);
+                        }
+                    });
+
+                    return someChecksFailed;
+                }).some(function(isCheckFailedForPlatform) {
+                    return isCheckFailedForPlatform;
                 });
 
-                return someChecksFailed;
-            }).some(function (isCheckFailedForPlatform) {
-                return isCheckFailedForPlatform;
+                if (someChecksFailed) throw new CordovaError('Some of requirements check failed');
             });
-
-            if (someChecksFailed) throw new CordovaError('Some of requirements check failed');
-        }).done();
```

Comment:
Looks like you end up removing "done()" - does that not cause us to lose exceptions?